javadocs

Jo-Hunter opened the pull request:
Created javadocs comments. Have not turned it into the html because it won't let me because of a lack of the library. 
Pushed the .gitignore by accident... not sure what changed or why.
Confused by recordDefinition interface having a method that returns a recordDefinition object.
Hope they are useful to you.

Jo-Hunter committed:
added javadoc comments

Jo-Hunter committed:
Merge branch 'master' of https://github.com/Jo-Hunter/SDgen

AussieGuy0 commented:
Awesome job, mate! 👍

> Have not turned it into the html because it won't let me because of a lack of the library.

Don't worry about not turning it into HTML, Maven handles all that.

> Pushed the .gitignore by accident... not sure what changed or why.

Looks like `/bin/` was added to the `.gitignore`. If you use Eclipse, it might have added it. In the future, make sure you're only pushing the correct files. I don't really care that it's there in this PR though.

> Confused by recordDefinition interface having a method that returns a recordDefinition object.

The purpose of this is so you can use addField() multiple times. For instance `Gen.start().addField("bla", new StringGenerator()).addField("blu", new StringGenerator()`. If it didn't have that, users could only have a single field!

Thanks for the help, this PR will be merged! I didn't actually need FieldData to be commented but the more docs the better, haha. I would highly recommend learning a bit about Maven, it'll make your life easier in the future. IDE's like Eclipse and Intellij have a lot of support for Maven.

Jo-Hunter commented:
So happy to have my first PR!!!! 
Thanks!

I will take your feedback on board. I added the files individually but I guess I thought commit all (.) would just commit the files I had added.
Yeah I have had Maven on my list of a million things to learn, probably time to get on board with that :) 

Thanks again! :)

AussieGuy0 commented:
I think I can see your problem. After your run `git add file1.java file2.java`, you only need to write `git commit` (no parameters). This will automatically commit all files that your previously added.

Summary:
1. `git add file1.java file2.java`
2. `git commit`
3. `git push`

If you have any more questions, don't be afraid to contact me (details on my website).
